src/pages/projects/create-project-modal/CreateProjectModal.tsx

```diff
@@ -43,8 +43,14 @@ const CreateProjectModal = (props: Props) => {
         testId='create-project-modal-submit-button'
         value='Create project'
         onClick={() => { 
-          props.setState(ModalState.CLOSED);
-          props.onSubmit(newProjectName);
+          props.setState(ModalState.LOADING);
+          
+          props.onSubmit(newProjectName).then(() => {
+            props.setState(ModalState.CLOSED);
+          }).catch(error => {
+            console.log(error);
+            props.setState(ModalState.INPUT);
+          });
```

Comment:
I think this whole thing should be inside `onSubmit` callback. So that modal is only (as much as its possible) presentation layer